### ISSUE TITLE
TransformControls: Give initial transform, in particular initial rotation

### ISSUE THREAD

**Issue by asfg84** [mrdoob/three.js#19436]
I would like to have a possibility to set an initial transform when creating a TransformControls object. Especially for the rotation: Instead of always starting with a rotation of (0, 0, 0) I would like to be able to give an initial rotation, so that scaling is applied in a rotated coordinate frame.
 
Example: Transform an object (mesh) using TransformControls and store its transformed position, size and orientation in a file. Then, when loading the transformed object from a file and displaying it, I want to continue transforming it in the same way using TransformControls. For this, I need a way to set the initial transform.

**Comment by gkjohnson** [comment]
> so that scaling is applied in a rotated coordinate frame.

Three.js cannot apply scaling in a rotated coordinate frame -- a Mesh can only be transformed along its cardinal axes. If you need a scale applied on a non-cardinal axis you'll need to create a parent transform and adjust that or modify the geometry vertices themselves.

Also if you need a custom rotated frame to move in you can create a proxy object representing the appropriate frame and apply the adjustments to your target object.

**Comment by asfg84** [comment]
I could solve it by adding an initialQuaternion property to TransformControls and in the attach() function I set 
```
this.object.quaternion.x = this.initialQuaternion[ 0 ];
this.object.quaternion.y = this.initialQuaternion[ 1 ];
this.object.quaternion.z = this.initialQuaternion[ 2 ];
this.object.quaternion.w = this.initialQuaternion[ 3 ];
```
By doing so, I can still translate and rotate the object around the world axes (using setSpace("world")), but apply the scaling transform in the rotated frame.

**Comment by arodic** [comment]
I don't understand this issue well. It sounds like it is concerning world space transformations only. Can you achieve this by using local space transformation mode?

Also, I don't see how it is possible to apply scale in a rotated frame in threejs that is why TransformControls does scale only in local space.

**Comment by asfg84** [comment]
I figured out that my problem was caused by using drei and not TransformControls directly. So I will close this issue.

When I use the following code, scaling of the rotated box is applied in its local frame, which is what I want:

```
const material = new THREE.MeshBasicMaterial( { color: "#ffff00" } );
const geometry = new THREE.BoxBufferGeometry( 0.4, 0.2, 0.2 );
const box = new THREE.Mesh(geometry, material);

box.quaternion.x = quat.x;
box.quaternion.y = quat.y;
box.quaternion.z = quat.z;
box.quaternion.w = quat.w;

scene.add( box );
const transformControl = new TransformControls( camera, renderer.domElement );
transformControl.attach( box );
scene.add( transformControl );
transformControl.setMode( "scale" );
```

However, I used fiber to create the box and then nested it inside the TransformControls from drei. In this case, the quaternion that I applied to the mesh for the box is not on the top level of the object attached to the TransformControls but is nested inside groups and arrays. Thus, the top-level object always had a unit quatermion and scaling was applied in the local frame of this top-level object (which is the same as the world frame in case of a unit rotation), although my mesh was rotated. Thus, I had a weird scaling behavior.
